src/Files.App/Services/NetworkDrivesService.cs

```diff
@@ -147,39 +150,51 @@ public Task OpenMapNetworkDriveDialogAsync()
 		/// <inheritdoc/>
 		public async Task<bool> AuthenticateNetworkShare(string path)
 		{
-			var netRes = new NETRESOURCE()
+			var netRes = new NETRESOURCEW() { dwType = NET_RESOURCE_TYPE.RESOURCETYPE_DISK };
+
+			unsafe
 			{
-				dwType = NETRESOURCEType.RESOURCETYPE_DISK,
-				lpRemoteName = path
-			};
+				fixed (char* lpcPath = path)
+					netRes.lpRemoteName = new PWSTR(lpcPath);
+			}
 
 			// If credentials are saved, this will return NO_ERROR
-			Win32Error connectionError = WNetAddConnection3(HWND.NULL, netRes, null, null, 0);
+			var res = (WIN32_ERROR)PInvoke.WNetAddConnection3W(new(nint.Zero), netRes, null, null, 0);
```

Comment:
I'd not change the variable name

Reply:
Convention is hResult, res, ret in Win32API, also thus enum also contains non-error code so I think this is good.